lang/java/reef-applications/reef-vortex/src/test/java/org/apache/reef/vortex/test/DefaultVortexManagerTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.test;
+
+import org.apache.reef.vortex.driver.DefaultVortexManager;
+import org.apache.reef.vortex.driver.VortexManager;
+import org.apache.reef.vortex.driver.scheduler.RandomTaskletScheduler;
+import org.apache.reef.vortex.driver.tasklet.*;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test DefaultVortexManager by simulating events using mocked classes
+ * TODO: test possible race conditions
+ */
+public class DefaultVortexManagerTest {
+  private TestUtil testUtil = new TestUtil();
+
+  @Test
+  public void testSingleTaskletNoFailure() throws Exception {
```

Comment:
Add comments to the tests. For each test, give a concise description on what you're testing.